src/Options/ParseOptions.hpp

```diff
@@ -431,6 +446,14 @@ struct get_impl<Tag, Metavariables, Tag> {
     return t;
   }
 };
+
+template <typename Metavariables>
+struct get_impl<InputSource, Metavariables, InputSource> {
+  template <typename OptionList, typename Group>
+  static InputSource::type apply(const Parser<OptionList, Group>& opts) {
```

Comment:
Should this return by const-ref?

Reply:
The function wrapping this forwards it on by value, so no point.